### PR TITLE
test: add fuzz test for create table

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)
#3174

## What's changed and what's your intention?

Add a fuzz test for create table

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: tests-fuzz/src/error.rs [hunk 1]
```diff
@@ -38,4 +38,11 @@ pub enum Error {
 
     #[snafu(display("No droppable columns"))]
     DroppableColumns { location: Location },
+
+    #[snafu(display("Failed to execute query"))]
+    ExecuteQuery {
```
Comment (MichaelScofield) [inline tests-fuzz/src/error.rs:43]: Better carry the query in the error msg.

File: tests-fuzz/src/generator.rs [hunk 2]
```diff
@@ -57,7 +57,13 @@ macro_rules! impl_random {
     ($type: ident, $value:ident, $values: ident) => {
         impl<R: Rng> Random<$type, R> for $value {
             fn choose(&self, rng: &mut R, amount: usize) -> Vec<$type> {
-                $values.choose_multiple(rng, amount).cloned().collect()
+                let mut result = std::collections::BTreeSet::new();
```
Comment (MichaelScofield) [inline tests-fuzz/src/generator.rs:60]: If you gonna shuffle the result, why use BTreeSet?

File: .github/actions/fuzz-test/action.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+          using: composite
+          steps:
+          - name: Run Fuzz Test
+            shell: bash
+            run: cargo fuzz run ${{ inputs.target }} --fuzz-dir tests-fuzz -D -s none -- -max_total_time=120
+            env:
+              GT_MYSQL_ADDR: 127.0.0.1:4002
+        EOF
+    - name: Run
+      uses: ./dynamic_fuzz_test
```
Comment (fengjiachun) [inline .github/actions/fuzz-test/action.yaml:25]: Add an empty line?

File: tests-fuzz/README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+```bash
+cargo fuzz fmt fuzz_target .crash --fuzz-dir tests-fuzz  
+```
+Rerun the fuzz test with the input.
+
+```bash
+cargo fuzz run fuzz_target .crash --fuzz-dir tests-fuzz
+```
+For more details, visit [cargo fuzz](https://rust-fuzz.github.io/book/cargo-fuzz/tutorial.html) or run the command `cargo fuzz --help`.
```
Comment (fengjiachun) [inline tests-fuzz/README.md:41]: Add an empty line?

**Review by fengjiachun — APPROVED** [review]

File: .github/actions/fuzz-test/action.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+name: Fuzz Test
+description: 'Fuzz test given setup and service'
+inputs:
+  target:
+    description: "The fuzz target to test"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Fuzz Test
```
Comment (MichaelScofield) [inline .github/actions/fuzz-test/action.yaml:10]: Why generate the fuzz test action dynamically?

**Comment by MichaelScofield** [comment]
Do we have the doc on "how to write fuzz test"? @WenyXu

**Comment by WenyXu** [comment]
> Do we have the doc on "how to write fuzz test"? @WenyXu

Yes, I will add one.

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**